examples/hawkbit-device-simulator/src/main/java/org/eclipse/hawkbit/simulator/amqp/SpReceiverService.java

```diff
@@ -82,28 +89,58 @@ public SpReceiverService(final RabbitTemplate rabbitTemplate, final AmqpProperti
      */
     @RabbitListener(queues = "${hawkbit.device.simulator.amqp.receiverConnectorQueueFromSp}", containerFactory = "listenerContainerFactory")
     public void recieveMessageSp(final Message message, @Header(MessageHeaderKey.TYPE) final String type,
-            @Header(MessageHeaderKey.THING_ID) final String thingId,
+            @Header(name = MessageHeaderKey.THING_ID, required = false) final String thingId,
             @Header(MessageHeaderKey.TENANT) final String tenant) {
-        checkContentTypeJson(message);
-        delegateMessage(message, type, thingId, tenant);
-    }
-
-    private void delegateMessage(final Message message, final String type, final String thingId, final String tenant) {
         final MessageType messageType = MessageType.valueOf(type);
 
         if (MessageType.EVENT.equals(messageType)) {
+            checkContentTypeJson(message);
             handleEventMessage(message, thingId);
             return;
         }
 
         if (MessageType.THING_DELETED.equals(messageType)) {
+            checkContentTypeJson(message);
             repository.remove(tenant, thingId);
             return;
         }
 
+        if (MessageType.PING_RESPONSE.equals(messageType)) {
+            final String correlationId = new String(message.getMessageProperties().getCorrelationId(),
+                    StandardCharsets.UTF_8);
+            openPings.remove(correlationId);
```

Comment:
I would check the return value of `remove` and in case we got a response with a correlationId which didn't exists I would probably log and error?
```
if(!openPings.remove(correlationId)){
  //TODO: Log an error 
}
```